File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        String temp = String.valueOf(x);
        int sum = 0;
        for(int i = 0; i < temp.length(); i++){
            sum += (int)(temp.charAt(i)-'0');
        }
        if(x % sum == 0){
            return true;
        }
        else{
            return false;
        }
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cs
public class Solution {
    public bool solution(int x) {
        string temp = x.ToString();
        int sum = 0;

        for (int i = 0; i < temp.Length; i++) {
            sum += temp[i] - '0';
        }

        return x % sum == 0;
    }
}